components/host-dashboard/ScheduledExpensesBanner.js

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useQuery } from '@apollo/client';
+import { create, Mode } from '@transferwise/approve-api-action-helpers';
+import { FormattedMessage } from 'react-intl';
+
+import { addAuthTokenToHeader } from '../../lib/api';
+import { API_V2_CONTEXT, gqlV2 } from '../../lib/graphql/helpers';
+
+import ConfirmationModal from '../ConfirmationModal';
+import { expensesListAdminFieldsFragment, expensesListFieldsFragment } from '../expenses/graphql/fragments';
+import { Box, Flex } from '../Grid';
+import TransferwiseIcon from '../icons/TransferwiseIcon';
+import MessageBox from '../MessageBox';
+import StyledButton from '../StyledButton';
+import { TOAST_TYPE, useToasts } from '../ToastProvider';
+
+const scheduledExpensesQuery = gqlV2/* GraphQL */ `
+  query HostDashboardScheduledExpenses(
+    $hostId: String!
+    $limit: Int!
+    $status: ExpenseStatusFilter
+    $payoutMethodType: PayoutMethodType
+  ) {
+    expenses(host: { id: $hostId }, limit: $limit, status: $status, payoutMethodType: $payoutMethodType) {
+      totalCount
+      offset
+      limit
+      nodes {
+        ...ExpensesListFieldsFragment
+        ...ExpensesListAdminFieldsFragment
+      }
+    }
+  }
+
+  ${expensesListFieldsFragment}
+  ${expensesListAdminFieldsFragment}
+`;
+
+const ScheduledExpensesBanner = ({ host, onSubmit, secondButton }) => {
+  const scheduledExpenses = useQuery(scheduledExpensesQuery, {
+    variables: { hostId: host.id, limit: 100, status: 'SCHEDULED_FOR_PAYMENT', payoutMethodType: 'BANK_ACCOUNT' },
+    context: API_V2_CONTEXT,
+  });
+  const { addToast } = useToasts();
+  const [showConfirmationModal, setConfirmationModalDisplay] = React.useState(false);
+
+  const hasScheduledExpenses = scheduledExpenses.data?.expenses?.totalCount > 0;
+  if (!hasScheduledExpenses) {
+    return null;
+  }
+
+  const request = create({ mode: process.env.WISE_ENVIRONMENT === 'production' ? Mode.PRODUCTION : Mode.SANDBOX });
+  const handlePayBatch = async () => {
+    const expenseIds = scheduledExpenses.data.expenses.nodes.map(e => e.id);
+    try {
+      await request(`${process.env.WEBSITE_URL}/api/services/transferwise/pay-batch`, {
```

Comment:
Why not using the GraphQL API here?

Reply:
This is already Wise's library in play.

Reply:
I'm not sure to fully understand the design here. Do you mean that `/api/services/transferwise/pay-batch` is expected to be called by both the Frontend and Transferwise's webhook?

Reply:
Sorry about my half-ass explanation, @Betree . 🤭
The Wise request lib is responsible for requesting our backend (using REST endpoint) and, if necessary present Wise's OTP frame and re-try the request with the necessary token information.
The necessary condition here is:
1. frontend lib requests our endpoint
2. our endpoint requests Wise
3. Wise requests OTP (returns 403 with an OTT Header
4. our endpoint proxies this error back to the frontend library
5. library does its thingy and request again our endpoint library with an OTT token
6. our endpoint requests Wise with the OTT token
7. Wise funds the transfer. 🎉

Reply:
I see, thanks for clarifying! It does make sense to have this as a REST endpoint then. It could be useful to write these details somewhere, but I'm not sure where.